### PR TITLE
Initial checkin for test app

### PR DESCRIPTION
**Pull request by deanesmith**
App used by Jenkins X BDD testing as a stable, predictable app for the sole purpose of automated testing.

### CODE REVIEW

**Review by daveconde — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Contribution guidelines
+
+## Filing issues
```
Comment (hferentschik) [inline CONTRIBUTING.md:3]: Looks like a double heading here

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Contribution guidelines
```
Comment (hferentschik) [inline CONTRIBUTING.md:1]: Do we really need CONTRIBUTING for a internal testing app?

File: charts/jx-app-test-lifecycle/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+description: A Helm chart for Kubernetes
```
Comment (hferentschik) [inline charts/jx-app-test-lifecycle/Chart.yaml:2]: We should start putting a proper description here.

File: charts/jx-app-test-lifecycle/templates/rolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+kind: RoleBinding
+metadata:
+  name: {{ template "fullname" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ template "fullname" . }}
+subjects:
+- kind: ServiceAccount
+  name: {{ template "fullname" . }}
```
Comment (hferentschik) [inline charts/jx-app-test-lifecycle/templates/rolebinding.yaml:11]: Missing new line

File: charts/jx-app-test-lifecycle/templates/serviceaccount.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
```
Comment (hferentschik) [inline charts/jx-app-test-lifecycle/templates/serviceaccount.yaml:8]: Missing new line

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      GIT_PROVIDER      = 'github.com'
+      CHARTMUSEUM_CREDS = credentials('jenkins-x-chartmuseum')
+    }
+    stages {
+      stage('CI Build and push snapshot') {
+        when {
+          branch 'PR-*'
+        }
+        environment {
+          PREVIEW_VERSION = "0.0.0-SNAPSHOT-$BRANCH_NAME-$BUILD_NUMBER"
```
Comment (hferentschik) [inline Jenkinsfile:16]: The preview variables are not needed

File: charts/preview/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+description: A Helm chart for Kubernetes
```
Comment (hferentschik) [inline charts/preview/Chart.yaml]: No need for any of the _preview_ files. We neither build nor deploy a preview.

File: charts/preview/values.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    - --cleanup
+  Annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: hook-succeeded
+
+preview:
+  image:
+    repository:
+    tag:
+    pullPolicy: IfNotPresent
```
Comment (hferentschik) [inline charts/preview/values.yaml]: newline

File: go.sum [hunk 9]
```diff
@@ -0,0 +1,142 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (hferentschik) [inline go.sum:1]: that seems to be an awful long list, given the short list if dependencies. Have you tried to delete it and just running make to let go modules re-create it.

File: charts/jx-app-test-lifecycle/Makefile [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+	sed -i "" -e "s/version:.*/version: $(VERSION)/" Chart.yaml
+	sed -i "" -e "s/tag: .*/tag: $(VERSION)/" values.yaml
+else ifeq ($(OS),Linux)
+	sed -i -e "s/version:.*/version: $(VERSION)/" Chart.yaml
+	sed -i -e "s/tag: .*/tag: $(VERSION)/" values.yaml
+else
+	echo "platfrom $(OS) not supported to tag with"
+	exit -1
+endif
+	git add --all
```
Comment (hferentschik) [inline charts/jx-app-test-lifecycle/Makefile]: This is not working. This builds are not reproducible this way. The commit to master which the tag is referring to is never pushed. See also https://github.com/jenkins-x-apps/jx-app-jacoco/issues/68. It is working now in jx-app-jacoco.

**Review by hferentschik — CHANGES_REQUESTED** [review]

**Comment by hferentschik** [comment]
@daveconde - I don't think prow is integrated yet ;-) This requires an entry here https://github.com/jenkins-x/prow-config/blob/master/prow/config.yaml

**Review by hferentschik — COMMENTED** [review]
Seems there is no .gitignore

File: Jenkinsfile (hunk 6 above)
Comment (deanesmith) [reply Jenkinsfile:16]: Removed

File: charts/jx-app-test-lifecycle/Chart.yaml (hunk 3 above)
Comment (deanesmith) [reply charts/jx-app-test-lifecycle/Chart.yaml:2]: Done

File: charts/preview/Chart.yaml (hunk 7 above)
Comment (deanesmith) [reply charts/preview/Chart.yaml]: Done

File: go.sum (hunk 9 above)
Comment (deanesmith) [reply go.sum:1]: Yes.  Both of these files are fresh and I tried once again to be sure.

File: charts/jx-app-test-lifecycle/Makefile (hunk 10 above)
Comment (deanesmith) [reply charts/jx-app-test-lifecycle/Makefile]: @hferentschik Will reach out to you regarding this.

Comment (hferentschik) [reply charts/jx-app-test-lifecycle/Makefile]: I recommend we align with cb-app-sor.

File: charts/jx-app-test-lifecycle/templates/service.yaml [hunk 11]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+kind: Service
```
Comment (hferentschik) [inline charts/jx-app-test-lifecycle/templates/service.yaml:2]: No need really for a service

**Comment by hferentschik** [comment]
@deanesmith have you pushed an update to this pull request already?

**Pull request merged**